ocode_python/core/semantic_context.py

```
@@ -0,0 +1,779 @@
+
+try:
+    # Try to import sentence-transformers for semantic embeddings
+    from sentence_transformers import SentenceTransformer
+
+    EMBEDDINGS_AVAILABLE = True
+except ImportError:
+    # Fallback to basic semantic analysis
+    EMBEDDINGS_AVAILABLE = False
+    SentenceTransformer = None  # type: ignore
+
+import numpy as np
```

Pylint (3.3.7)

[error] Unable to import 'numpy'

(E0401)